packages/core/src/hooks/use-call.ts

```diff
@@ -121,13 +121,34 @@ function queryKey({
   args,
   blockIdentifier,
 }: { chain?: Chain; contract?: Contract } & CallArgs) {
+  function convertBigIntsToStrings(input: any): any {
+    if (typeof input === "bigint") {
+      return input.toString(10);
+    }
+    if (Array.isArray(input)) {
+      // Recursively process each element in the array
+      return input.map((item) => convertBigIntsToStrings(item));
+    }
+    if (input && typeof input === "object") {
+      // If it's an object, process each key-value pair
+      const result: { [key: string]: any } = {};
+      for (const key in input) {
+        if (Object.prototype.hasOwnProperty.call(input, key)) {
+          result[key] = convertBigIntsToStrings(input[key]);
+        }
+      }
+      return result;
+    }
+
+    return input;
+  }
   return [
     {
       entity: "readContract",
       chainId: chain?.name,
       contract: contract?.address,
       functionName,
-      args,
+      args: convertBigIntsToStrings(args),
```

Comment:
How about serializing the value to string? 

```suggestion
      args: Json.stringify(args, (_, v) => typeof v === "bigint" ? v.toString() : v)
```

Reply:
Thanks, that's much better! I will use that!